### ISSUE TITLE
even with packing=False SFTTrainer uses ConstantLengthDataset

### ISSUE THREAD

**Issue by kforcreation** [huggingface/trl#631]
even with packing=False SFTTrainer is using ConstantLengthDataset
![image](https://github.com/huggingface/trl/assets/59497032/d8ab9f5d-b76a-47b3-a600-95aa004de3ab)

**Comment by mallorbc** [comment]
I installed TRL from the source in edit mode.  

I print out the train dataset with and without packing on the imdb dataset.

Without packing I see:
Dataset({
    features: ['input_ids', 'attention_mask'],
    num_rows: 25000
})

With packing I see:
<trl.trainer.utils.ConstantLengthDataset object at 0x7f8e7cd33a00>

**Comment by mallorbc** [comment]
So with packing false, the correct behavior is individual entries with padding right?

With packing set to true, it fills the context window from multiple examples?

So one is good for finetuning(the first), and one is good for pretraining?

**Comment by kforcreation** [comment]
True
True
True ( not very sure though, people keep coming up with new ideas )
